src/textual/widgets/_data_table.py

```diff
@@ -569,77 +878,93 @@ def clear(self, columns: bool = False) -> None:
         Args:
             columns: Also clear the columns. Defaults to False.
         """
-        self.row_count = 0
         self._clear_caches()
         self._y_offsets.clear()
         self.data.clear()
         self.rows.clear()
         if columns:
             self.columns.clear()
-        self._line_no = 0
         self._require_update_dimensions = True
-        self.cursor_cell = Coordinate(0, 0)
-        self.hover_cell = Coordinate(0, 0)
+        self.cursor_coordinate = Coordinate(0, 0)
+        self.hover_coordinate = Coordinate(0, 0)
         self.refresh()
 
-    def add_columns(self, *labels: TextType) -> None:
-        """Add a number of columns.
-
-        Args:
-            *labels: Column headers.
-        """
-        for label in labels:
-            self.add_column(label, width=None)
-
-    def add_column(self, label: TextType, *, width: int | None = None) -> None:
+    def add_column(
+        self, label: TextType, *, width: int | None = None, key: str | None = None
+    ) -> ColumnKey:
         """Add a column to the table.
 
         Args:
             label: A str or Text object containing the label (shown top of column).
-            width: Width of the column in cells or None to fit content. Defaults to None.
-        """
-        text_label = Text.from_markup(label) if isinstance(label, str) else label
+            width: Width of the column in cells or None to fit content.
+            key: A key which uniquely identifies this column.
+                If None, it will be generated for you.
 
-        content_width = measure(self.app.console, text_label, 1)
+        Returns:
+            Uniquely identifies this column. Can be used to retrieve this column
+                regardless of its current location in the DataTable (it could have moved
+                after being added due to sorting/insertion/deletion of other columns).
+        """
+        column_key = ColumnKey(key)
```

Comment:
Should we check for duplicates here? I can imagine users forgetting and making two calls with the same `key`.